lib/Controller/PushController.php

```diff
@@ -111,7 +112,15 @@ public function registerDevice($pushTokenHash, $devicePublicKey) {
 		}
 
 		$deviceIdentifier = hash('sha512', json_encode([$user->getCloudId(), $token->getId()]));
-		openssl_sign($deviceIdentifier, $signature, $key->getPrivate(), OPENSSL_ALGO_SHA512);
+
+		$privateKey = new RSA();
+		$privateKey->setPrivateKey($key->getPrivate(), RSA::PRIVATE_FORMAT_PKCS1);
+		$privateKey->setSignatureMode(RSA::SIGNATURE_PSS);
+		$privateKey->setHash('sha512');
+		$privateKey->setMGFHash('sha512');
```

Comment:
I also tried without `setHash()`, without `setMGFHash()` and without both. But all signatures don't verify in go